executor/instruction/mod.rs

```diff
@@ -344,15 +344,18 @@ fn type_from_row_or_annotations<'a>(
     vertex: &Vertex<ExecutorVariable>,
     row: MaybeOwnedRow<'_>,
     annos: impl Iterator<Item = &'a Type> + fmt::Debug,
-) -> Type {
+) -> Option<Type> {
     match vertex {
         &Vertex::Variable(ExecutorVariable::RowPosition(var)) => {
             debug_assert!(row.len() > var.as_usize());
-            let VariableValue::Type(type_) = row.get(var).to_owned() else { unreachable!("Supertype must be a type") };
-            type_
+            match row.get(var).to_owned() {
+                VariableValue::Type(type_) => Some(type_),
+                VariableValue::Empty => None,
```

Comment:
I explicitly avoided making this return optional when I introduced the `Unsatisfiable` constraint.
Do we need to do it now? Should we not be generating answers if it's not set?

Reply:
Answering myself again:
With dynamic modes it's likely we'll only ever call this with the value bound. So I'll just do an unwrap outside for now and see if all calls here are followed by an unwrap